sys/ufs/ffs/fs.h

```diff
@@ -256,7 +257,13 @@
 struct fsck_cmd {
 	int32_t	version;	/* version of command structure */
 	int32_t	handle;		/* reference to filesystem to be changed */
-	int64_t	value;		/* inode or block number to be affected */
+#ifdef __ILP32__
```

Comment:
A possibly cleaner way to fix this upstream would be to make 'value' a union of a uint64_t and a uintptr_t.  I don't think Kirk used an int64_t to avoid freebsd32 compat, but to instead be that union.  That would mean the diff downstream in CheriBSD would be solely to support hybrid kernels.

Reply:
I thought about this, but then you need to pad the unintptr_t on 32-bit BE systems and you can't trust that the high-bits are zeroed. I'll think about this more.

Reply:
Mmm.  I think you can ignore the upper bits, but it's true you'd have a bit more of a PITA for big-endian (though we are down to only one left: powerpc).  You could use an anon struct to make this work, something evil like:

```
    union {
        uint64_t number;
        struct {
#if BIG_ENDIAN && ILP32
             uint32_t dummy;
#endif
             uintptr_t pointer;
       };
    };
```

for the upstream case.  For CHERI we'd have the added bit to use uintcap_t for the _KERNEL case.

Reply:
But maybe that's not really cleaner than what you've done.

Reply:
I've got kintcap_t working on FreeBSD build hosts. We'll see if I need further changes for Linux. That's at least a bit cleaner.

Reply:
Ugh, it looks like this is going to be a mess. I'd need to add x86/_types.h on non-freebsd x86 and it looks like a number of guards aren't compatible with MacOS and would have to be made so. :(

Reply:
The double underscore version only needs types.h

Reply:
Hm I see...

Reply:
Simply not defining the structure when building makefs seems to do the trick. There's probably an argument it should be `#if defined(_KERNEL) || (WANT_FSCK_CMD)` with a corresponding update to fsck.